data-service/src/integTest/java/com/amazonaws/blox/dataservice/api/DeploymentIntegTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.dataservice.api;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import com.amazonaws.blox.dataservice.Application;
+import com.amazonaws.blox.dataservice.model.DeploymentStatus;
+import com.amazonaws.blox.dataservice.model.DeploymentType;
+import com.amazonaws.blox.dataservicemodel.v1.client.DataService;
+import com.amazonaws.blox.dataservicemodel.v1.exception.DeploymentDoesNotExist;
+import com.amazonaws.blox.dataservicemodel.v1.exception.EnvironmentExistsException;
+import com.amazonaws.blox.dataservicemodel.v1.exception.EnvironmentNotFoundException;
+import com.amazonaws.blox.dataservicemodel.v1.exception.EnvironmentVersionOutdatedException;
+import com.amazonaws.blox.dataservicemodel.v1.exception.InvalidParameterException;
+import com.amazonaws.blox.dataservicemodel.v1.exception.ServiceException;
+import com.amazonaws.blox.dataservicemodel.v1.model.CreateEnvironmentRequest;
+import com.amazonaws.blox.dataservicemodel.v1.model.CreateEnvironmentResponse;
+import com.amazonaws.blox.dataservicemodel.v1.model.GetDeploymentRequest;
+import com.amazonaws.blox.dataservicemodel.v1.model.GetDeploymentResponse;
+import com.amazonaws.blox.dataservicemodel.v1.model.InstanceGroup;
+import com.amazonaws.blox.dataservicemodel.v1.model.StartDeploymentRequest;
+import com.amazonaws.blox.dataservicemodel.v1.model.StartDeploymentResponse;
+import java.util.UUID;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@ContextConfiguration(classes = {Application.class})
+public class DeploymentIntegTest {
+
+  @Autowired private DataService dataService;
+
+  @Test
+  public void createAndGetDeployment()
```

Comment:
This is a good start to test that everything is wired up, but in order to evaluate the data model, we need a more realistic outline of how we'd interact with the Deployment resource. Could we simulate the calls that the scheduling manager would make in a test case here, so that we can ensure the data model serves that purpose appropriately?

In particular, these are ones that stand out from the [design](https://github.com/blox/blox/blob/dev/docs/daemon_design.md#pending-deployments):
- how does the scheduling manager reconcile deployments?
- what queries will the scheduling manager make when new deployments show up?
- how will the scheduling workflow update the deployment state?
- how will the UI display ongoing deployments?